### PR TITLE
Introduce django-s3sign for media upload - MEDT-3915

### PR DESCRIPTION
**Pull request by nikolas**
Basic image upload and asset creation flow is in place.

* Added new UploadedAssetCreateView in assetmgr

Screenshot_2021-09-08_13-48-28

### CODE REVIEW

**Comment by nikolas** [comment]
This is only failing due to the master branch failing, resolved here: https://github.com/ccnmtl/mediathread/pull/3997

File: mediathread/assetmgr/views.py [hunk 1]
```diff
@@ -324,6 +328,61 @@ def post(self, request):
...
+            width = request.POST.get('width')
+            height = request.POST.get('height')
+
+        Source.objects.create(
+            asset=asset, url=url,
+            primary=True,
+            width=width, height=height,
+            label=label)
+
+        asset_url = reverse('asset-view', args=[
```
Comment (sdreher) [inline mediathread/assetmgr/views.py:374]: Before returning, a global annotation needs to be created to ensure the asset shows up in the user's collection.

`asset.global_annotation(user, True)`

**Review by sdreher — COMMENTED** [review]
This feature is working well. One change needed to ensure the uploaded item lands in the user's collection (see comments). We'll definitely want to review the design before we push this live. Nice work!

**Comment by nikolas** [comment]
This branch has been updated with the following changes:

* Add "1" and "2" icons
* On upload complete, hide upload progress
* Hook up https://www.npmjs.com/package/bs-custom-file-input
Screenshot_2021-09-08_13-48-28

I'm looking into updating the style of the Browse button, but this branch can also be merged as is. @sdreher

File: mediathread/assetmgr/views.py [hunk 2]
```diff
@@ -324,6 +328,62 @@ def post(self, request):
...
+            request.course.pk, asset.pk
+        ])
+
+        messages.success(
+            request,
+            'Asset created: <strong><a href="{}">{}</a></strong>'.format(
```
Comment (sdreher) [inline mediathread/assetmgr/views.py:381]: Update confirmation message: `The <a href ="asset_url"><strong>asset.title</strong> item has been added to your collection.

File: mediathread/assetmgr/views.py [hunk 3]
```diff
@@ -324,6 +328,62 @@ def post(self, request):
...
+    def post(self, request, *args, **kwargs):
+        if (not request.POST.get('title')) or \
+           (not request.POST.get('url')):
+            return HttpResponseBadRequest(
+                'Title and URL are required to make an asset.')
+
+        title = request.POST.get('title').strip()
+        url = request.POST.get('url')
+
+        asset = Asset.objects.create(
```
Comment (sdreher) [inline mediathread/assetmgr/views.py:355]: Check for the "Upload As" parameter and ensure the author is being set as specified.

**Review by sdreher — COMMENTED** [review]
The interface updates look very good. I've added two comments to address, then I'm happy to merge.

**Review by sdreher — COMMENTED** [review]
One more note as you demo'd -
* Can you change the button name from `Create Asset` to `Upload`

**Pull request merged**